rules-reviewed/eap7/eap6/resteasy.windup.xml

```diff
@@ -0,0 +1,744 @@
+                    <link title="RESTEasy SPI Application Changes"
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_resteasy_deprecated_classes" />
+                    <link title="JAX-RS API" href="https://jax-rs-spec.java.net/nonav/2.0-rev-a/apidocs/index.html?javax/ws/rs/container/ContainerResponseFilter.html" />
+                    <tag>resteasy</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="resteasy-eap6-000030">
+            <when>
+                <javaclass references="org.jboss.resteasy.spi.interception.PreProcessInterceptor" />
+            </when>
+            <perform>
+                <hint title="Deprecated PreProcessInterceptor interface" effort="1" severity="mandatory">
+                    <message>Use `javax.ws.rs.container.ContainerRequestFilter` instead of `org.jboss.resteasy.spi.interception.PreProcessInterceptor`.
+                    The RESTEasy interceptor facility introduced in release 2.x is replaced by the JAX-RS 2.0 compliant filters and interceptor facility in release 3.0.x.</message>
+                    <link title="RESTEasy SPI Application Changes"
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_resteasy_deprecated_classes" />
+                    <link title="JAX-RS API" href="https://jax-rs-spec.java.net/nonav/2.0-rev-a/apidocs/index.html?javax/ws/rs/container/ContainerRequestFilter.html" />
+                    <tag>resteasy</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="resteasy-eap6-000032">
+            <when>
+                <javaclass references="org.jboss.resteasy.spi.Registry" />
+            </when>
+            <perform>
+                <hint title="RESTEasy SPI Application Changes" effort="1" severity="mandatory">
+                    <message>There were multiple changes on adding/removing methods on `org.jboss.resteasy.spi.Registry`.</message>
+                    <link title="RESTEasy SPI Application Changes"
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/chapter-4-application-changes#injectorfactory_and_registry" />
+                    <link title="RESTEasy SPI Registry Changes"
+                        href="https://docs.jboss.org/resteasy/docs/3.0.13.Final/javadocs/index.html?org/jboss/resteasy/spi/Registry.html" />
+                    <tag>resteasy</tag>
+                </hint>
+            </perform>
+        </rule>        
+        <!-- Exceptions -->
+        <rule id="resteasy-eap6-000101">
+            <when>
+                <javaclass references="org.jboss.resteasy.spi.ForbiddenException" />
```

Comment:
This seems to only exist in RESTEasy 3. But it's true that it's deprecated, so I think we can keep the rule.